### PR TITLE
fixes #58 added callstack information for uprobes

### LINKED ISSUE #58: Callstacks on Uprobes
On Windows you can right-click on a probe of a dynamically instrumented function and get the callstack of that call. This should be also possible in the Linux and remote version.

### PR DESCRIPTION
**Pull request by florian-kuebler**

### CODE REVIEW

File: OrbitCore/BpfTrace.cpp [hunk 1]
```diff
@@ -103,14 +105,97 @@ uint64_t BpfTrace::ProcessString(const std::string& a_String)
...
+        std::string function = tokens[1];
+
+        std::string moduleRaw = tokens[2];
+        std::string module = Replace(moduleRaw.substr(1), ")\n", "");
+
+        // TODO: this is copy&paste from LinuxPerf.cpp
+        std::wstring moduleName = ToLower(Path::GetFileName(s2ws(module)));
+        std::shared_ptr<Module> moduleFromName = Capture::GTargetProcess->GetModuleFromName( ws2s(moduleName) );
+        // Debug - Temporary
+        if( Contains(function, "Renderer::render") )
```
Comment (pierricgimmig) [inline OrbitCore/BpfTrace.cpp]: Please remove the temporary debug code.

File: OrbitCore/BpfTrace.cpp [hunk 2]
```diff
@@ -103,14 +105,97 @@ uint64_t BpfTrace::ProcessString(const std::string& a_String)
...
+        {
+            PRINT_VAR(function);
+            PRINT_VAR(moduleName);
+            PRINT_VAR(address);
+        }
+        if( moduleFromName )
+        {
+            uint64_t new_address = moduleFromName->ValidateAddress(address);
+            // Debug - Temporary
```
Comment (pierricgimmig) [inline OrbitCore/BpfTrace.cpp]: Please remove.

File: OrbitCore/BpfTrace.cpp [hunk 3]
```diff
@@ -103,14 +105,97 @@ uint64_t BpfTrace::ProcessString(const std::string& a_String)
...
+            if( Contains(function, "Renderer::render") )
+            {
+                std::cout << std::hex << address << " -> " << std::hex << new_address << std::endl;
+            }
+            address = new_address;
+        }
+        else
+        {
+            // Debug - Temporary
```
Comment (pierricgimmig) [inline OrbitCore/BpfTrace.cpp]: Please remove.

**Comment by florian-kuebler** [comment]
> Thanks for this Florian, it will be very useful. I'm curious to know more about performance implications of the bpftrace symbol resolution. Until we have our own solution for getting callstacks and doing symbol resolution, this will do just fine. We could expose a toggle for callstack collection if it ends up eating up too much cycles.

Hi Pierric,
I will have a look in the performance implications ASAP. If there are serious ones, we should go for the toggle until we have our own solution implemented. Also I would assume that bpftrace relies on framepointers for stack unwinding (such as perf in its default mode). In general this is a restriction that would not be necessary.

**Pull request merged**